server/routes/api/lob.js

```diff
@@ -18,27 +18,146 @@ const DELIVERABILITY_WARNINGS = {
+      throw new Error(
+        'Address object must include both city and state, or a ZIP code'
+      )
+    }
+  } catch (validationError) {
+    return res.status(400).send({ error: validationError.message })
+  }
+
+  const { line1, line2, city, state, zip } = address
+  // Ensure the ZIP code is at least 5 digits
+  const zipCode = zip ? zip.padStart(5, '0') : null
+
+  try {
+    const lob = new Lob({ apiKey: getLobApiKey() })
+    const response = await lob.usVerifications.verify({
+      primary_line: line1,
+      secondary_line: line2,
+      city,
+      state,
+      zip_code: zipCode
+    })
+
+    const {
+      deliverability,
+      primary_line: revisedLine1,
+      secondary_line: revisedLine2,
+      components: {
+        city: revisedCity,
+        state: revisedState,
+        zip_code: revisedZip,
+        zip_code_plus_4: revisedZipPlus4,
+        address_type: addressType,
+        record_type: recordType
+      }
+    } = response
+
+    const isUndeliverable =
+      !deliverability || deliverability === 'undeliverable'
+    const isResidential = addressType === 'residential'
+    const isPostOfficeBox = recordType === 'po_box'
+    const isPuertoRico = revisedState === 'PR'
+
+    const deliverable =
+      !isUndeliverable && isResidential && !isPostOfficeBox && !isPuertoRico
+    const warning = DELIVERABILITY_WARNINGS[deliverability] || null
+
+    if (!deliverable) {
+      let errorMessage = 'Address is undeliverable'
+      if (!isUndeliverable) {
+        if (!isResidential) {
+          errorMessage = 'Non-residential addresses are not currently supported'
+        } else if (isPostOfficeBox) {
+          errorMessage = 'Post office boxes are not currently supported'
+        } else if (isPuertoRico) {
+          errorMessage = 'Puerto Rico addresses are not currently supported'
+        }
+      }
+
+      return res.status(400).send({ error: errorMessage })
+    }
+
+    revisedAddress = {
+      deliverable,
+      warning,
+      line1: revisedLine1,
+      line2: revisedLine2 || null,
+      city: revisedCity,
+      state: revisedState,
+      zip: revisedZip + (revisedZipPlus4 ? '-' + revisedZipPlus4 : '')
+    }
+  } catch (error) {
+    // This endpoint should not return anything other than `200` status
+    // codes, even for undeliverable addresses
+    return handleLobError(error, res)
+  }
+
   try {
+    // Create Lob address using variables passed into route via post body
+    const addressResponse = await Lob.addresses.create({
+      name: req.body.name,
+      address_line1: revisedAddress.line1,
+      address_line2: revisedAddress.line2,
+      address_city: revisedAddress.city,
+      address_state: revisedAddress.state,
+      address_zip: revisedAddress.zip,
+      address_country: 'US'
+    })
+
+    // Create letter using lob library
     await lob.letters.create(
       {
         description,
         to,
```

Comment:
I like these changes.

Is the `to` being in as an address `id` from lob?

Reply:
I see that `from` (the user's address) is being processed fine.
I don't see any address processing for the `to` field (the representative's address)

Reply:
Hey @nawazkhan I'm finishing this up, so writing test to confirm I properly understand this. I believe it is inferred from req.body in line 33 

` const { description, to, from, template_id } = req.body || {}`